romancal/tweakreg/tweakreg_step.py

```diff
@@ -0,0 +1,560 @@
+                if ALIGN_TO_ABS_REFCAT:
+                    # NOTE: This .name attrib agreed upon by the JWST Cal
+                    #       Working Group.
+                    #       Current value is merely a place-holder based
+                    #       on HST conventions. This value should also be
+                    #       translated to the FITS WCSNAME keyword
+                    #       IF that is what gets recorded in the archive
+                    #       for end-user searches.
+                    imcat.wcs.name = f"FIT-LVL3-{self.abs_refcat}"
+
+                image_model.meta.wcs = imcat.wcs
+
+        return images
+
+    def _is_wcs_correction_small(self, wcs, twcs):
+        """Check that the newly tweaked wcs hasn't gone off the rails"""
+        tolerance = 10.0 * self.tolerance * u.arcsec
+
+        ra, dec = wcs.footprint(axis_type="spatial").T
+        tra, tdec = twcs.footprint(axis_type="spatial").T
+        skycoord = SkyCoord(ra=ra, dec=dec, unit="deg")
+        tskycoord = SkyCoord(ra=tra, dec=tdec, unit="deg")
+
+        separation = skycoord.separation(tskycoord)
+
+        return (separation < tolerance).all()
+
+    def _imodel2wcsim(self, image_model):
+        image_model = image_model if isinstance(
+            image_model, datamodels.DataModel
+            ) else datamodels.open(path.basename(image_model))
+        # make sure that we have a catalog:
+        if hasattr(image_model, 'catalog'):
+            catalog = image_model.catalog
+        else:
+            catalog = image_model.meta.tweakreg_catalog
+
+        model_name = path.splitext(image_model.meta.filename)[0].strip('_- ')
+
+        if isinstance(catalog, Table):
+            if not catalog.meta.get('name', None):
+                catalog.meta['name'] = model_name
+        else:
+            try:
+                cat_name = str(catalog)
+                catalog = Table.read(catalog)
+                catalog.meta['name'] = cat_name
+            except OSError:
+                self.log.error(f"Cannot read catalog {catalog}")
+
+        # make sure catalog has 'x' and 'y' columns
+        for axis in ['x', 'y']:
+            if axis not in catalog.colnames:
+                long_axis = axis + 'centroid'
+                if long_axis in catalog.colnames:
+                    catalog.rename_column(long_axis, axis)
+                else:
+                    raise ValueError(
+                        "'tweakreg' source catalogs must contain either "
+                        "columns 'x' and 'y' or 'xcentroid' and "
+                        "'ycentroid'."
+                    )
+
+        # create WCSImageCatalog object:
+        refang = image_model.meta.wcsinfo
+        # TODO: create RSTWCSCorrector in tweakwcs
+        im = JWSTWCSCorrector(
+            wcs = image_model.meta.wcs,
+            wcsinfo = {'roll_ref': refang['roll_ref'],
+                     'v2_ref': refang['v2_ref'],
+                     'v3_ref': refang['v3_ref']},
```

Comment:
nitpick: the alignment does not look right here

Reply:
Fixed.